packages/twenty-server/src/engine/api/graphql/workspace-query-runner/workspace-query-runner.service.ts

```diff
@@ -305,11 +298,61 @@ export class WorkspaceQueryRunnerService {
     return parsedResults;
   }
 
+  async upsertMany<Record extends IRecord = IRecord>(
+    args: CreateManyResolverArgs<Record>,
+    options: WorkspaceQueryRunnerOptions,
+  ): Promise<Record[] | undefined> {
+    const results = await Promise.all(
+      args.data.map(async (payload) => {
+        const existingRecord = payload.id
+          ? await this.findOne(
```

Comment:
Should we use findMany with `args.data.map((payload) => payload.id)` instead of n findOne and store this in a map earlier for better performances?